packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -21,7 +21,7 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
   /// Creates an object that fetches the image at the given URL.
   ///
   /// The arguments [url] and [scale] must not be null.
-  const NetworkImage(this.url, { this.scale = 1.0, this.headers })
+  const NetworkImage(this.url, { this.scale = 1.0, this.headers, this.headerResolver })
```

Comment:
It seems a little confusing to me to have both the `headers` and the `headerResolver`, particularly since the HTTP spec allows for duplicate header keys (even though our implementation doesn't).

Adding this API seems a little scary to me. `NetworkImage` is already pretty problematic in terms of how it handles/respects/doesn't respect network headers, caching, etc.

Is there some reason to not just write a better provider?

Reply:
In our case, we would have to completely reimplement `NetworkImage`, since our current solution is prone to breakage when `ImageStreamCompleter` gains new functionality. For example, we don't handle listeners unsubscribing and resubscribing, and there's not a lot we can do because we can't handle disposal ourselves.

Perhaps it would be better to assert if both are provided? Instead of merging the headers.

Reply:
From offline discussion: it seems like it should be possible to wrap NetworkImage with a custom provider to avoid this.